crates/tauri-macros/src/command/handler.rs

```diff
@@ -31,33 +31,108 @@ pub struct Handler {
+
+    let mut command_defs = input
+      .parse_terminated(CommandDef::parse, Token![,])?
+      .into_iter()
+      .collect();
+
+    filter_unused_commands(plugin_name, &mut command_defs);
+    let mut commands = Vec::new();
+    let mut wrappers = Vec::new();
 
     // parse the command names and wrappers from the passed paths
-    let (commands, wrappers) = command_defs
-      .iter()
-      .map(|command_def| {
-        let mut wrapper = command_def.path.clone();
-        let last = super::path_to_command(&mut wrapper);
+    for command_def in &command_defs {
+      let mut wrapper = command_def.path.clone();
+      let last = super::path_to_command(&mut wrapper);
 
-        // the name of the actual command function
-        let command = last.ident.clone();
+      // the name of the actual command function
+      let command = last.ident.clone();
 
-        // set the path to the command function wrapper
-        last.ident = super::format_command_wrapper(&command);
+      // set the path to the command function wrapper
+      last.ident = super::format_command_wrapper(&command);
 
-        (command, wrapper)
-      })
-      .unzip();
+      commands.push(command);
+      wrappers.push(wrapper);
+    }
 
     Ok(Self {
-      command_defs: command_defs.into_iter().collect(), // remove punctuation separators
+      command_defs,
       commands,
       wrappers,
     })
   }
 }
 
+/// Try to get the plugin name by parsing the input for a `#![plugin(...)]` attribute,
+/// if it's not present, try getting it from `CARGO_PKG_NAME` enviroment variable
+fn try_get_plugin_name(input: &ParseBuffer<'_>) -> Result<Option<String>, syn::Error> {
+  if let Ok(attrs) = input.call(Attribute::parse_inner) {
+    for attr in attrs {
+      if attr.path().is_ident("plugin") {
+        // Parse the content inside #![plugin(...)]
+        let plugin_name = attr.parse_args::<Ident>()?.to_string();
+        return Ok(Some(if plugin_name == "__TAURI_CHANNEL__" {
+          plugin_name
+        } else {
+          plugin_name.replace("_", "-")
+        }));
+      }
+    }
+  }
+  Ok(
+    std::env::var("CARGO_PKG_NAME")
+      .ok()
+      .and_then(|var| var.strip_prefix("tauri-plugin-").map(String::from)),
+  )
+}
+
+fn filter_unused_commands(plugin_name: Option<String>, command_defs: &mut Vec<CommandDef>) {
+  let allowed_commands = tauri_utils::acl::read_allowed_commands();
+  let Some(allowed_commands) = allowed_commands else {
+    return;
+  };
+
+  let mut unused_commands = Vec::new();
+
+  let command_prefix = if let Some(plugin_name) = &plugin_name {
+    format!("plugin:{plugin_name}|")
+  } else {
+    // All application commands are allowed if we don't have an application ACL
+    //
+    // note that inline plugins without the #![plugin()] attribute would also get to this check
+    // which means inline plugins must have an app manifest to get proper unused command removal
+    if !allowed_commands.has_app_acl {
```

Comment:
i would move this if out of this block as it's a side effect while we're just getting the command prefix